codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -410,19 +412,30 @@ public static Http2Headers toHttp2Headers(HttpHeaders inHeaders, boolean validat
         return out;
     }
 
+    private static CharSequenceMap<AsciiString> toLowercaseMap(List<String> values) {
+        CharSequenceMap<AsciiString> result = new CharSequenceMap<AsciiString>();
+        for (CharSequence value : values) {
```

Comment:
consider using an "old for" loop if the List implements `RandomAccess` to reduce GC pressure (as the foreach thingy you use here will always create an Iterator)

Reply:
While it creates an iterator, it can be allocated on the stack. (Whether it _actually_ is allocated on the stack is a different discussion.)

Reply:
@ejona86 we saw GC issues caused by this in the past so better safe then sorry ;)

Reply:
looks like it's a linked list: https://github.com/netty/netty/blob/4.1/codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java#L163